src/agent/market_hours.rs

```diff
@@ -0,0 +1,353 @@
+            "Saturday",
+            "Sunday",
+        ] {
+            let mhkind_str = split_by_commas.next().ok_or(anyhow!(
+                "Market hours schedule ends before mandatory {} field",
+                weekday
+            ))?;
+
+            let mhkind: MHKind = mhkind_str.trim().parse().context(format!(
+                "Could not parse {} field from {:?}",
+                weekday, mhkind_str
+            ))?;
+
+            weekday_schedules.push(mhkind);
+        }
+
+        // We expect specifying wrong (incl. too large) amount of days
+        // to be an easy mistake. We should catch it to avoid acting
+        // on ambiguous schedule when there's too many day schedules
+        // specified.
+        if let Some(one_too_many) = split_by_commas.next() {
+            return Err(anyhow!("Found unexpected 8th day spec {:?}", one_too_many));
+        }
+
+        // The compiler was not too happy with moving values via plain [] access
+        let mut weekday_sched_iter = weekday_schedules.into_iter();
+
+        let result = Self {
+            timezone: tz,
+            // These unwraps failing would be an internal error, but
+            // panicking here does not seem wise.
+            mon:      weekday_sched_iter
+                .next()
+                .ok_or(anyhow!("INTERNAL: weekday_sched_iter too short"))?,
+            tue:      weekday_sched_iter
+                .next()
+                .ok_or(anyhow!("INTERNAL: weekday_sched_iter too short"))?,
+            wed:      weekday_sched_iter
+                .next()
+                .ok_or(anyhow!("INTERNAL: weekday_sched_iter too short"))?,
+            thu:      weekday_sched_iter
+                .next()
+                .ok_or(anyhow!("INTERNAL: weekday_sched_iter too short"))?,
+            fri:      weekday_sched_iter
+                .next()
+                .ok_or(anyhow!("INTERNAL: weekday_sched_iter too short"))?,
+            sat:      weekday_sched_iter
+                .next()
+                .ok_or(anyhow!("INTERNAL: weekday_sched_iter too short"))?,
+            sun:      weekday_sched_iter
+                .next()
+                .ok_or(anyhow!("INTERNAL: weekday_sched_iter too short"))?,
+        };
+
+        if let Some(_i_wish_lol) = weekday_sched_iter.next() {
+            Err(anyhow!("INTERNAL: weekday_sched_iter too long"))
+        } else {
+            Ok(result)
+        }
+    }
+}
+
+/// Helper enum for making clear time range, all-day open and all-day closed distinction
+#[derive(Debug, Eq, PartialEq)]
+pub enum MHKind {
+    Open,
+    Closed,
+    TimeRange(NaiveTime, NaiveTime),
+}
+
+impl MHKind {
+    pub fn can_publish_at(&self, when_market_local: NaiveTime) -> bool {
+        dbg!(&when_market_local);
+        dbg!(self);
+        match self {
+            Self::Open => true,
+            Self::Closed => false,
+            Self::TimeRange(start, end) => start <= &when_market_local && &when_market_local <= end,
```

Comment:
I think you need to support 24:00 in the format so that you can have intervals like:

17:00-24:00,00:00-17:00 (for 5pm on the first day through 5pm on the second day).

I think this current check will stop publishing between 11:59 pm and midnight every day if you're not careful.

Reply:
also can you add a test case for this day wraparound behavior?